### PR TITLE
Update CI to publish docker image

### PR DESCRIPTION
**Pull request by lebaudantoine**
## Purpose

Publish production Docker images for staging deployment.

## Proposal

I have followed [the basic usage of these configurations](https://github.com/docker/build-push-action).
Images are pushed to this [public Docker Hub repository](https://hub.docker.com/repository/docker/lasuite/people/general).

It closes, #39 .

- [x] Create and share a team Docker Hub account.
- [x] Update Sops keys.
- [x] Enable CI for automatic Docker image publication upon merging into main.

### CODE REVIEW

File: .github/workflows/docker-hub.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      -
+        name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        run: echo "$DOCKER_HUB_PASSWORD" | docker login -u "$DOCKER_HUB_USER" --password-stdin
+      -
+        name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
```
Comment (sampaccoud) [inline .github/workflows/docker-hub.yml:42]: so I understand you run the job on pull requests and do this in order to still test building the image even if you don't publish it.
In this case, shouldn't we remove the `  build-docker` job as well?

Comment (lebaudantoine) [reply .github/workflows/docker-hub.yml:42]: Indeed, yes, I could remove the build-docker `job` wdyt?

File: .github/workflows/docker-hub.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Docker Hub Workflow
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
```
Comment (AntoLC) [inline .github/workflows/docker-hub.yml:9]: Is is still necessary? 
Is it for staging, right? We want all the branches merged to main to be pushed to dockerhub, isn't it ?

File: .github/workflows/docker-hub.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        run: echo "$DOCKER_HUB_PASSWORD" | docker login -u "$DOCKER_HUB_USER" --password-stdin
+      -
+        name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
```
Comment (AntoLC) [inline .github/workflows/docker-hub.yml:43]: It will increment automatically the version of the docker images, is it that ?

**Review by AntoLC — COMMENTED** [review]
🥇  Looks good and straightforward.

Could we create a staging branch or whatever the name to plug this workflow to try it ?

File: .github/workflows/docker-hub.yml (hunk 3 above)
Comment (lebaudantoine) [reply .github/workflows/docker-hub.yml:43]: I followed [this example](https://github.com/docker/metadata-action?tab=readme-ov-file#basic). They gave some example on how tags will behave.

File: .github/workflows/docker-hub.yml (hunk 2 above)
Comment (lebaudantoine) [reply .github/workflows/docker-hub.yml:9]: I might be wrong, but this job will be used to push Docker Images, in any situation. It should run when:
- pushes to the 'main' branch
- pushes to tags starting with 'v'
- pull requests targeting the 'main' branch (to run Docker build without push).

**Review by AntoLC — APPROVED** [review]
🚀

**Review by rouja — APPROVED** [review]
LGTM,
Do not  forget to cleanup this image if you do not use it : https://hub.docker.com/layers/lasuite/people/pr-40/images/sha256-cd1a93e3a04a9122bab215afbc6bbc4d2eebeffde38419dc793d2a471a5b7ec1?context=explore

**Pull request merged**